_SPIKE_.md

```diff
@@ -0,0 +1,244 @@
+# Booter for creating REST APIs from model files
+
+Quoting from https://github.com/strongloop/loopback-next/issues/2036:
+
+> In LoopBack 3, it was very easy to get a fully-featured CRUD REST API with
+> very little code: a model definition describing model properties + a model
+> configuration specifying which datasource to use.
+>
+> Let's provide the same simplicity to LB4 users too.
+>
+> - User creates a model class and uses decorators to define model properties.
+>   (No change here.)
+> - User declaratively defines what kind of data-access patterns to provide
+>   (CRUD, KeyValue, etc.) and what datasource to use under the hood.
+> - `@loopback/boot` processes this configuration and registers appropriate
+>   repositories & controllers with the app.
+
+In this Spike, I am demonstrating a PoC implementation of an extensible booter
+that processed model configuration files in JSON formats and uses 3rd-party
+plugins to build repository & controller classes at runtime.
+
+## Basic use
+
+Create `src/public-models` directory in your project. For each model you want to
+expose via REST API, add a new `.config.ts` file that's exporting the model
+configuration.
+
+Example (`src/public-models/product.config.ts`):
```

Comment:
We can use `product.resource.ts` or `product.endpoint.ts` once we decide the proper name.

Reply:
+1

Personally, I'd use `product.rest-config.ts` and if we add support for gRPC controllers later, then `product.grpc-config.ts`.